sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/AnalysisHelper.scala

```diff
@@ -243,13 +243,22 @@ trait AnalysisHelper extends QueryPlan[LogicalPlan] { self: LogicalPlan =>
    *               subtree. Do not pass it if the rule is not purely functional and reads a
    *               varying initial state for different invocations.
    */
-  def resolveExpressionsWithPruning(cond: TreePatternBits => Boolean,
```

Comment:
If you look at other methods in this file, they usually have 3 variants:
1. no up and down in the name, which calls the down version
2. the down version
3. the up version

We should follow it here.

Reply:
Good point. Updated to follow the same style as `resolveOperators`.